stdlib/dynarray.ml

```diff
@@ -264,17 +267,25 @@ end = struct
         a
       else copy_from_array a ~dummy
 
-    let unsafe_nocopy_to_array a =
-      if Array.length a = 0 || Obj.(tag (repr a.(0)) <> double_tag) then
-        a
-      else begin
-        let n = Array.length a in
-        let a' = Array.make n a.(0) in
-        for i = 1 to n - 1 do
-          Array.unsafe_set a' i (unsafe_get (Array.unsafe_get a i))
-        done;
-        a'
-      end
+    exception Dummy_found of int
+
+    let unsafe_nocopy_to_array a ~dummy =
+      let arr =
+        if Array.length a = 0 || Obj.(tag (repr a.(0)) <> double_tag) then
+          a
+        else begin
+          let n = Array.length a in
+          let a' = Array.make n a.(0) in
+          for i = 1 to n - 1 do
+            Array.unsafe_set a' i (unsafe_get (Array.unsafe_get a i))
+          done;
+          a'
+        end
+      in
+      Array.iteri
+        (fun i v -> if is_dummy v dummy then raise (Dummy_found i))
+        arr;
```

Comment:
I would recommend writing a loop directly here instead of `Array.iteri`, which is going to be sensibly slower and to allocate a closure, at least if `flambda` is not used.

Reply:
@OlivierNicole I notice that this comment has not been addressed, is it an explicit choice on your end?